src/v/cloud_storage/remote_partition.cc

```diff
@@ -647,18 +646,25 @@ std::unique_ptr<remote_segment_batch_reader> remote_partition::borrow_reader(
   storage::log_reader_config config, model::offset key, segment_state& st) {
     if (std::holds_alternative<offloaded_segment_state>(st)) {
         gc_stale_materialized_segments(true);
+    } else if (_readers_limit.current() <= 0) {
+        vlog(_ctxlog.trace, "Reader limit pressure: force-GCing all segments");
+        gc_stale_materialized_segments(true);
```

Comment:
The code in `gc_stale_materialized_segments`: https://github.com/redpanda-data/redpanda/blob/ee28fc924d7431214dfaa88854317b8bd208b0cc/src/v/cloud_storage/remote_partition.cc#L415 
seems to imply that this call to gc will do the opposite of what the comment says? 

If `force` is true then the `max_idle` is 60 seconds (opposed to 0ms), and the chance of a materialized segment state not being gc'd increases? I think it might be an issue with the way `max_idle` is set, it might need to be the opposite ie

```
    auto max_idle = force_collection ?  0ms : stm_max_idle_time;
```

Reply:
this is now fixed in dev so just needs a rebase